src/functional/ffront/dialect_parser.py

```diff
@@ -110,11 +112,13 @@ def apply(
     ) -> DialectRootT:  # type: ignore[valid-type]  # used to work, now mypy is going berserk for unknown reasons
 
         source, filename, starting_line = source_definition
+        line_offset = starting_line - 1
         try:
             definition_ast = ast.parse(textwrap.dedent(source)).body[0]
             definition_ast = RemoveDocstrings.apply(definition_ast)
             definition_ast = FixMissingLocations.apply(definition_ast)
             definition_ast = ast.increment_lineno(definition_ast, starting_line - 1)
+            line_offset -= starting_line - 1
```

Comment:
```suggestion
            definition_ast = ast.increment_lineno(definition_ast, starting_line - 1)
            definition_ast = FixMissingLocations.apply(definition_ast)
            line_offset -= starting_line - 1
```
I realised during the code-walk that the order matters here. Otherwise, `ast.increment_lineno` won't fix the line locations of the nodes where `FixMissingLocations` just added missing locations (see: https://github.com/python/cpython/blob/e9ac890c0273aee413aa528cc202c3efa29f1d7a/Lib/ast.py#L240 ).